js/src/utils/toScopeState.test.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Internal dependencies
+ */
+import toScopeState from './toScopeState';
+
+const SCOPE = {
+	// Manage product listings and accounts for Google Shopping
+	CONTENT: 'https://www.googleapis.com/auth/content',
+	// Manage new site verifications with Google
+	SITE_VERIFICATION_VERIFY_ONLY:
+		'https://www.googleapis.com/auth/siteverification.verify_only',
+	// Manage AdWords campaigns
+	AD_WORDS: 'https://www.googleapis.com/auth/adwords',
+};
+
+describe( 'toScopeState', () => {
+	let contentScopes;
+	let siteVerificationScopes;
+	let gmcScopes;
+	let adsScopes;
+	let allScopes;
+
+	beforeEach( () => {
+		contentScopes = [ SCOPE.CONTENT ];
+		siteVerificationScopes = [ SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		gmcScopes = [ SCOPE.CONTENT, SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		adsScopes = [ SCOPE.AD_WORDS ];
+		allScopes = [
+			SCOPE.CONTENT,
+			SCOPE.SITE_VERIFICATION_VERIFY_ONLY,
+			SCOPE.AD_WORDS,
+		];
+	} );
```

Comment:
❔ How about this without needing `beforeEach`? 🤔 

```suggestion
	const contentScopes = [ SCOPE.CONTENT ];
	const siteVerificationScopes = [ SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
	const gmcScopes = [ SCOPE.CONTENT, SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
	const adsScopes = [ SCOPE.AD_WORDS ];
	const allScopes = [
		SCOPE.CONTENT,
		SCOPE.SITE_VERIFICATION_VERIFY_ONLY,
		SCOPE.AD_WORDS,
	];
```

Reply:
Note: if we do this, we can even move them to the top level, outside of the `describe` block.

Reply:
Quoted from https://github.com/woocommerce/google-listings-and-ads/pull/834#discussion_r663177789:

> Maybe I'm paranoid, but I'd move those into `beforeEach` as we would re-use it in every test

This style is based on Tomek's suggestion, but I don't have a particular preference for it. Perhaps you could have a 📜  with @tomalec, and sum up a team convention. 🤔

Reply:
I don't have really strong arguments for it. 
Personally, I like to reason about it like:
- If I need something to happen/to be used for `it` blocks within `describe`. Like initialize variables, define helper function, define "cases", to programmatically generate test, etc.
	I put that inside `describe`. For me, that's something needed to construct `it` blocks, not the actual tests, test's contents.
- If I need something to be prepared for actual tests but once for all. Like, to start a mock server.
	I use `before`. However, I use it relatively rarely.
- If I set up or need something to be prepared for every test content, for every tested behavior. Like, initialize properties/state for a component, mock individual response. So something that will be used **inside** the `it` block.
	I use `beforeEach`.

I believe that makes it easier to reason about the code, as you can anticipate, how the variable & co. will be used just by their location, without inspecting every usage below.
It makes the tests easier to maintain as if you add or remove `it` block, you would not have to adjust the scope of the setup code.
It makes tests more stable, as it helps to make the test less inter-connected and inter-dependent. So one test is not able to mess with the setup of another one.